File: samples/CslaLight/cs/NavigationSample/NavigationApp/CustomControlFactory.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

namespace NavigationApp
{
  public class CustomControlFactory : Csla.Silverlight.IControlNameFactory
  {

    #region IControlNameFactory Members

    public Control ControlNameToControl(string controlName)
    {
      if (controlName.ToUpper().Contains("_1".ToUpper()))
      {
        return new ControlOne();
      }
      else if (controlName.ToUpper().Contains("_2".ToUpper()))
      {
        return new ControlTwo();
      }
      else if (controlName.ToUpper().Contains("_5".ToUpper()))
      {
        return new ControlFive();
      }
      else
        return null;
    }

    public string ControlToControlName(Control control)
    {
      if (control is ControlOne)
        return "Control_1";
      if (control is ControlTwo)
        return "Control_2";
      if (control is ControlFive)
        return "Control_5";
      return control.GetType().AssemblyQualifiedName;
    }

    #endregion
  }
}

File: Samples/CslaLight/vb/NavigationSample/NavigationApp/CustomControlFactory.vb
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes

Namespace NavigationApp
  Public Class CustomControlFactory
    Implements Csla.Silverlight.IControlNameFactory

#Region "IControlNameFactory Members"

    Public Function ControlNameToControl(ByVal controlName As String) As Control Implements Csla.Silverlight.IControlNameFactory.ControlNameToControl
      If controlName.ToUpper().Contains("_1".ToUpper()) Then
        Return New ControlOne()
      ElseIf controlName.ToUpper().Contains("_2".ToUpper()) Then
        Return New ControlTwo()
      ElseIf controlName.ToUpper().Contains("_5".ToUpper()) Then
        Return New ControlFive()
      Else
        Return Nothing
      End If
    End Function

    Public Function ControlToControlName(ByVal control As Control) As String Implements Csla.Silverlight.IControlNameFactory.ControlToControlName
      If TypeOf control Is ControlOne Then
        Return "Control_1"
      End If
      If TypeOf control Is ControlTwo Then
        Return "Control_2"
      End If
      If TypeOf control Is ControlFive Then
        Return "Control_5"
      End If
      Return control.[GetType]().AssemblyQualifiedName
    End Function

#End Region
  End Class
End Namespace